From 9cf557216cbd0fe6e5ca235114c27260622d049b Mon Sep 17 00:00:00 2001
From: Bryan Kadzban <bz-glibc@kdzbn.homelinux.net>
Date: Thu, 11 Jun 2009 11:32:51 -0700
Subject: [PATCH 030/100] Fix futex syscall parameter for x86 absolute timeout waits.

This affects only installations compiled for kernels older than 2.6.18.
(cherry picked from commit 1828530f09bcc321cfb84c27110bbce804605ec4)
---
 nptl/ChangeLog                                     |    8 ++++++++
 .../unix/sysv/linux/i386/i486/lowlevellock.S       |    2 +-
 2 files changed, 9 insertions(+), 1 deletions(-)

2009-06-11  Ulrich Drepper  <drepper@redhat.com>

	[BZ #10262]
	* sysdeps/unix/sysv/linux/i386/i486/lowlevellock.S
	(LOAD_FUTEX_WAIT_ABS): Fix futex parameter in case private futexes
	cannot be assumed.
	Patch by Bryan Kadzban <bz-glibc@kdzbn.homelinux.net>.

diff --git a/nptl/sysdeps/unix/sysv/linux/i386/i486/lowlevellock.S b/nptl/sysdeps/unix/sysv/linux/i386/i486/lowlevellock.S
index 056b729..7578c7e 100644
--- a/nptl/sysdeps/unix/sysv/linux/i386/i486/lowlevellock.S
+++ b/nptl/sysdeps/unix/sysv/linux/i386/i486/lowlevellock.S
@@ -60,7 +60,7 @@
 # define LOAD_FUTEX_WAIT_ABS(reg) \
 	xorl	$FUTEX_PRIVATE_FLAG, reg ; \
 	andl	%gs:PRIVATE_FUTEX, reg ; \
-	orl	$FUTEX_WAIT | FUTEX_CLOCK_REALTIME, reg
+	orl	$FUTEX_WAIT_BITSET | FUTEX_CLOCK_REALTIME, reg
 # define LOAD_FUTEX_WAKE(reg) \
 	xorl	$FUTEX_PRIVATE_FLAG, reg ; \
 	andl	%gs:PRIVATE_FUTEX, reg ; \
-- 
1.6.5.2

